src/Symfony/Component/Translation/CacheTranslatorBag.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+use Symfony\Component\Config\ConfigCacheInterface;
+use Symfony\Component\Config\ConfigCacheFactoryInterface;
+use Symfony\Component\Config\ConfigCacheFactory;
+
+/**
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+class CacheTranslatorBag implements TranslatorBagInterface
+{
+    /**
+     * @var TranslatorBagInterface
+     */
+    private $translatorBag;
+
+    /**
+     * @var string|null
+     */
+    private $cacheDir;
+
+    /**
+     * @var bool
+     */
+    private $debug;
+
+    /**
+     * @var ConfigCacheFactoryInterface|null
+     */
+    private $configCacheFactory;
+
+    /**
+     * @param string|null                 $cacheDir           The directory to use for the cache
+     * @param bool                        $debug              Use cache in debug mode ?
+     * @param TranslatorBagInterface      $translatorBag
+     * @param ConfigCacheFactoryInterface $configCacheFactory
+     */
+    public function __construct($cacheDir, $debug = false, TranslatorBagInterface $translatorBag = null, ConfigCacheFactoryInterface $configCacheFactory = null)
+    {
+        $this->cacheDir = $cacheDir;
+        $this->debug = $debug;
+        $this->translatorBag = $translatorBag ?: new FallbackTranslatorBag();
+        $this->configCacheFactory = $configCacheFactory ?: new ConfigCacheFactory($this->debug);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCatalogue($locale = null)
```

Comment:
hmm I know about it, right now I'm not sure how to deal with it :), any thought ?
in `3.0` I would like to remove the default value:
```php
public function getCatalogue($locale)
```